lib/src/action.dart

```diff
@@ -41,7 +43,7 @@ import 'dart:async';
 /// when a consumer needs to check state changes immediately after invoking an
 /// action.
 ///
-class Action<T> implements Function {
```

Comment:
Mixing in and subclassing are equivalent when there's no other mixins or super classes present.